server/src/main/java/org/elasticsearch/index/mapper/Mapper.java

```diff
@@ -53,6 +59,10 @@ public Settings indexSettings() {
         public Version indexCreatedVersion() {
             return Version.indexCreated(indexSettings);
         }
+
+        public Mapper.ColumnPositionResolver getColumnPositionResolver() {
+            return columnPositionResolver;
+        }
```

Comment:
Instead of exposing the resolver I think this could expose the underlying methods directly. The ColumnPositionResolver seems like an implementation detail. https://en.wikipedia.org/wiki/Law_of_Demeter

The two column position methods could probably also be merged into one to encapsulate the `if /else` blocks scattered everywhere.

Reply:
Thanks for sharing the link